### ISSUE TITLE
REQUIRE and build.jl for unregistered packages

### ISSUE THREAD

**Issue by etotheipluspi** [JuliaPOMDP/POMDPs.jl#82]
My understanding is that POMDPs.jl will be the only JuliaPOMDP package registered with METADATA. 
All the other packages can be added using `POMDPs.add("package_name")`. `POMDPs.add("package_name")` checks if "package_name" is a JuliaPOMDP package, clones it and builds it. 

I'd like to propose that we use `POMDPs.add(...)` to add all unregistered package dependencies in the build.jl file. I noticed that some packages have the dependencies in the REQUIRE file, and that actually causes Julia to throw an error when building a package. 

For example, MCTS.jl needs GenerativeModels.jl and POMDPToolbox.jl. If the build file `deps/build.jl` looks like:

``` julia
using POMDPs
POMDPs.add("GenerativeModels")
POMDPs.add("POMDPToolbox")
```

Then when the user runs POMDPs.add("MCTS"), the two dependencies will get cloned and built automatically. However, if the REQUIRE file has GenerativeModels or POMDPToolbox in it, Julia will throw an error, because it will check if the requirements are met before doing a build. 

TLDR; Do not put unregistered dependencies in the REQUIRE file for JuliaPOMDP packages (they won't get installed anyway), put any dependencies in build.jl, and use `POMDPs.add`.  

Edit: please add any JuliaPOMDP packages needed to test your package to the `build.jl` file as well. For testing solvers, we usually need `POMDPModels`.

**Comment by tawheeler** [comment]
Good thoughts!
To clarify, registered dependencies still go in REQUIRE?

**Comment by etotheipluspi** [comment]
Yep, edited the post to clarify!

**Comment by etotheipluspi** [comment]
This is the adopted convention now, closing.
